sorc/grid_tools.fd/regional_esg_grid.fd/pesg.f90

```diff
@@ -62,59 +62,57 @@ module pesg
 
 contains
 
-!=============================================================================
+!> Inverse of xstoxc. I.e., cartesians to stereographic
+!! @author R. J. Purser
+!! @param xc xstoxc
+!! @param xs Inverse of xstoxc
 subroutine xctoxs(xc,xs)!                                             [xctoxs]
-!=============================================================================
-! Inverse of xstoxc. I.e., cartesians to stereographic
-!=============================================================================
 implicit none
 real(dp),dimension(3),intent(in ):: xc
 real(dp),dimension(2),intent(out):: xs
-!-----------------------------------------------------------------------------
 real(dp):: zp
-!=============================================================================
 zp=u1+xc(3); xs=xc(1:2)/zp
 end subroutine xctoxs
 
-!=============================================================================
+!> Standard transformation from polar stereographic map coordinates, xs, to
```

Comment:
At top of file, we need two doxygen documentation blocks, one for file, one for module. You need to insert a blank line.